### ISSUE TITLE
Question: HTTPS Protocol being lost with redirect

### ISSUE THREAD

**Issue by innes** [kipusoep/UrlTracker#128]
We have a large number of entries on the URL tracker and also an HTTP to HTTPS rewrite rule in web.config.  

We are seeing an issue causing redirect chains where the incoming request is being redirected to the correct node as specified in the URLTracker, but in doing so is losing the HTTPS protocol.  As a result when it requests the new node it then has a further 301 redirect from HTTP to HTTPS.  We cannot see why this first redirect would not honour the protocol of the original request.  

URLTracker Version 3.3
Umbraco Version 6.2.5

We would appreciate some guidance here please?

Thanks

Innes

**Comment by JayyajGH** [comment]
A bit late to the party but I've just come across the same issue.  Although I haven't proven it I've got an idea of why we are having the issue.

The code in the UrlTrackerModule uses the scheme of the request to set the scheme of the redirected response when redirecting to an Umbraco node.  Our webservers aren't secured end to end and therefore I think the webserver is seeing an HTTPS request as being HTTP and then using that for the response.

A better solution might be to just inherit the scheme by using \\<redirect-url> rather then explicity setting the scheme to HTTP or HTTPS.  If I get a chance I'll try and prove this and submit a PR.

**Comment by kipusoep** [comment]
Yes a reverse proxy (nginx or Azure for example) strips HTTPS from the request and might cause issues.

**Comment by JayyajGH** [comment]
It might be worth supporting one/all of the standard header fields for disclosing information lost in proxying so that the original scheme can be used.  Looks like there isn't a single standard though.

https://stackoverflow.com/questions/16042647/whats-the-de-facto-standard-for-a-reverse-proxy-to-tell-the-backend-ssl-is-used
